### PR TITLE
[CSL-2305] Use withCompressedFiles and tempDir for tarxz archive

### PR DESCRIPTION
**Pull request by volhovm**
Should fix the permission problem. Tested with report server 0.4.9.

### CODE REVIEW

File: infra/Pos/Reporting/Methods.hs [hunk 1]
```diff
@@ -144,20 +144,15 @@ retrieveLogFiles lconfig = fromLogTree $ lconfig ^. lcTree
...
     tar <- tarPackIndependently files
     tarxz <-
         BS.concat <$>
         runConduitRes (yield tar .| Lzma.compress (Just 0) .| consume)
-    aName <- getArchiveName
-    withFile aName WriteMode $ \handle -> do
-        BS.hPut handle tarxz
-        hFlush handle
-    pure aName
+    pure tarxz
```
Comment (gromakovsky) [inline infra/Pos/Reporting/Methods.hs]:
```
    a <- b
    pure a
```
is the same as
```
    b
```

**Comment by gromakovsky** [comment]
Meanwhile I will test it using the installer from https://github.com/input-output-hk/daedalus/pull/751 after the build finishes there.

File: infra/Pos/Reporting/Methods.hs (hunk 1 above)
Comment (volhovm) [reply infra/Pos/Reporting/Methods.hs]: I wanted to be explicit about what do I produce. It doesn't do any harm.

**Comment by volhovm** [comment]
Previously `targz` bs was dumped to the file in the logs directory, but now it uses temporary directory (`withCompressedLogs`), which should not cause any permissions problems. I assume that `getCanonicalTemporaryDirectory` returns something I can write to.

**Comment by dcoutts** [comment]
> Previously targz bs was dumped to the file in the logs directory, but now it uses temporary directory (withCompressedLogs), which should not cause any permissions problems.

Why was that failing? We can write to the logs dir, since that's where we write logs! What am I missing?

**Comment by volhovm** [comment]
@dcoutts I don't know the answer, but the error clearly indicated that I couldn't write the archive. Seems that it was a relative directory of where program was run, not a logs dir.

**Comment by volhovm** [comment]
@dcoutts I don't think you're precise here. This patch doesn't include any refactoring that wasn't directly needed. What I have done here is I separated file creation from `compressLogs` and created `withCompressedLogs` function that operates in CPS. The latter one is only 3 lines bigger in fact (not counting comments), because i needed to do `withTempFile` directly. Then I updated two usages of `compressLogs`.

**Comment by volhovm** [comment]
@dcoutts Fixed. https://github.com/input-output-hk/cardano-sl/pull/2591

**Comment by volhovm** [comment]
@gromakovsky also please review/merge #2591

**Comment by gromakovsky** [comment]
Leneengrad's street taught me one rule…

**Pull request merged**